users/dierkes/preprocessing/wav2vec.py

```diff
@@ -10,58 +11,206 @@ class FairseqAudioManifestCreationJob(Job):
-        self, audio_dir_path, file_extension="wav", valid_percent=0.01, seed=42, path_must_contain=None
+        self,
+        audio_dir_path,
+        file_extension="wav",
+        valid_percent=0.01,
+        seed=42,
+        path_must_contain=None,
+        upsampling_alpha=None,
+        manifest_audio_paths=None,
     ):
         """
-        :param tk.Path audio_dir_path: path to raw audio files to be included
+        :param [tk.Path]|tk.Path audio_dir_path: list of paths or single path to raw audio files to be included
         :param str file_extension: file extension to look for in audio_dir_path
         :param float valid_percent: percentage of files to be in validation set
         :param int seed: random seed for splitting into train and valid set
         :param str|None path_must_contain: if set, path must contain this substring
             for a file to be included in the manifest
+        :param float upsampling_alpha: specifies how much to upsample low resource languages. Upsampling
+            calculation is done according to "Unsupervised Cross-Lingual Representation Learning for Speech
+            Recognition", see  arXiv:2006.13979v2
+        :param [str]|str|None manifest_audio_paths: explicitly specifies output paths in manifest for each
+            audio directory respectively. Allows to use different paths in the manifest than in the audio_dir_paths
         """
-        self.audio_dir_path = audio_dir_path
+        if isinstance(audio_dir_path, tk.Path):
+            self.audio_dir_paths = [audio_dir_path]
+        else:
+            assert isinstance(audio_dir_path, list)
+            self.audio_dir_paths = audio_dir_path
+        assert min([isinstance(path, tk.Path) for path in self.audio_dir_paths])
+
+        if len(self.audio_dir_paths) == 1:
+            assert (
+                upsampling_alpha is None
+            ), "Only one audio directory is given, upsampling not possible"
+
+        if manifest_audio_paths:
+            if isinstance(manifest_audio_paths, tk.Path):
+                manifest_audio_paths = [manifest_audio_paths]
+            assert len(manifest_audio_paths) == len(self.audio_dir_paths)
+            self.manifest_audio_paths = [
+                os.path.realpath(path) for path in manifest_audio_paths
+            ]
+        else:
+            self.manifest_audio_paths = None
+
         self.file_extension = file_extension
         self.valid_percent = valid_percent
-        assert 0 <= self.valid_percent <= 1.0
+        assert 0.0 <= self.valid_percent <= 1.0
         self.seed = seed
         self.path_must_contain = path_must_contain
+        self.upsampling_alpha = upsampling_alpha
+        if self.upsampling_alpha is not None:
+            assert 0.0 <= self.upsampling_alpha <= 1.0
 
         self.out_manifest_path = self.output_path("manifest/", directory=True)
-        self.rqmt = {"time": 2, "mem": 8, "cpu": 1}
+        self.rqmt = {"time": 6, "mem": 8, "cpu": 1}
```

Comment:
does it make sense to make `time` configurable for the user?

Reply:
I am not sure actually. I think 6 hours should cover most of the use cases so I would make it configurable for the user. I tested the job with a combined amount of ~1100 hours of speech.